simul/matplotlib/Paper_cosi2/plotit.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python
+# Plots the graph of one of the test-runs
+# It takes the CSV-file as argument and shows the plot
+# of the times used for each round
+
+import os
+
+os.environ["LC_ALL"] = "en_US.UTF-8"
+os.environ["LANG"] = "en_US.UTF-8"
+
+import sys
+
+sys.path.insert(1, '..')
+from mplot import MPlot
+from stats import CSVStats
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches
+
+
+def plotData(data, name,
+             xlabel="Number of witnesses", ylabel="Seconds per round",
+             xticks=[], loglog=[2, 2], xname="hosts",
+             legend_pos="lower right",
+             yminu=0, ymaxu=0,
+             title=""):
+    mplot.plotPrepareLogLog(loglog[0], loglog[1])
+    plots = read_csvs_xname(xname, *data[0])
+    ranges = []
+    data_label = []
+    plot_show(name)
+
+    for index, label in enumerate(data[1]):
+        data_label.append([plots[index], label])
+        ranges.append(
+            mplot.plotMMA(plots[index], 'round_wall', colors[index][0], 4,
+                          dict(label=label, linestyle='-', marker='o',
+                               color=colors[index][1], zorder=5)))
+
+    # Make horizontal lines and add arrows for JVSS
+    xmin, xmax, ymin, ymax = CSVStats.get_min_max(*ranges)
+    if yminu != 0:
+        ymin = yminu
+    if ymaxu != 0:
+        ymax = ymaxu
+    plt.ylim(ymin, ymax)
+    # plt.xlim(16, xmax * 1.2)
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+
+    plt.legend(loc=legend_pos)
+    plt.axes().xaxis.grid(color='gray', linestyle='dashed', zorder=0)
+    if len(xticks) > 0:
+        ax = plt.axes()
+        ax.set_xticks([16, 32, 64, 128, 256, 512, 1024, 4096, 16384, 65536])
+    if title != "":
+        plt.title(title)
+    mplot.plotEnd()
+    return data_label
+
+
+# Plots a Cothority and a JVSS run with regard to their averages. Supposes that
+# the last two values from JVSS are off-grid and writes them with arrows
+# directly on the plot
+def plotRoundtime():
+    plotData([['jvss', 'naive_cosi', 'ntree_cosi', 'cosi_depth_3'],
+              ['JVSS', 'Naive', 'NTree', 'CoSi']], 'comparison_roundtime',
+             xticks=[4, 8, 16, 32, 64, 128, 256, 512, 1024, 4096, 16384, 65536],
```

Comment:
Is there a reason that there is only the jvss.csv pushed to the branch? Can't find the others (calling the python script will fail without redoing all the simulations...)